File: LeetCode/Algorithm_II/Day4/844_Backspace String Compare.py
class Solution:
    def get_result(self, s):
        stack = []
        for letter in s: 
            if letter == '#':
                if stack:
                    stack.pop()
            else:
                stack.append(letter)
        return ''.join(stack)
                
            
    def backspaceCompare(self, s: str, t: str) -> bool:
        return self.get_result(s) == self.get_result(t)
        

File: LeetCode/DayChallenges/DayChallenge_2022_May/844_Backspace_String_Compare.js
/**
 * @param {string} s
 * @return {string[]}
 */
const getWordStack = function (s) {
  const stack = [];
  for (let i = 0; i < s.length; i += 1) {
    if (s[i] !== '#') {
      stack.push(s[i]);
    } else {
      if (stack.length > 0) {
        stack.pop();
      }
    }
  }
  return stack;
};

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
const backspaceCompare = function (s, t) {
  const stackS = getWordStack(s);
  const stackT = getWordStack(t);
  if (stackS.length !== stackT.length) return false;
  for (let i = 0; i < s.length; i += 1) {
    if (stackS[i] !== stackT[i]) return false;
  }
  return true;
};